### PR TITLE
Debug rework

### PR DESCRIPTION
**Pull request by gaaf**
Migrate to esphome native logging

### CODE REVIEW

**Review by elupus — COMMENTED** [review]
Thanks for the change, and it does look good. I am not sure i want this however. The idea has been to keep that class arduino compatible. So it could become an arduino lib at some point.

**Comment by gaaf** [comment]
> The idea has been to keep that class arduino compatible. 

That ship has sailed already, pins and uart already use esphome code.

Debug logging as it is now is next to unusable.

**Comment by elupus** [comment]
True that!

File: components/nibegw/NibeGw.cpp [hunk 1]
```diff
@@ -106,25 +95,14 @@ void NibeGw::loop()
...
         byte b = RS485->read();
-
-#ifdef ENABLE_NIBE_DEBUG
-        if (debug)
-        {
-          sprintf(debug_buf, "%02X ", b);
-          debug(3, debug_buf);
-        }
-#endif
+        ESP_LOGVV(TAG, "%02X", b);
```
[the remark this answers is not in the document]
Comment (gaaf) [reply components/nibegw/NibeGw.cpp:98]: If I read the code correctly, it is compile disabled if log level > VERY_VERBOSE.

[the remark this answers is not in the document]
Comment (elupus) [reply components/nibegw/NibeGw.cpp:98]: Yes looks like it. I will see if i can delay logs somewhat until we are in less time critical locations for debug level stuff.

File: components/nibegw/NibeGw.cpp [hunk 2]
```diff
@@ -205,21 +173,12 @@ void NibeGw::loop()
         }
         else
         {
-#ifdef ENABLE_NIBE_DEBUG
-          if (debug)
-            debug(2, "No message to send\n");
-#endif
+          ESP_LOGD(TAG, "No message to send");
```
Comment (elupus) [inline components/nibegw/NibeGw.cpp:176]: This is likely the cause. We delay too long before sending ack. Probably didn't work with old debug enabled either.

Comment (gaaf) [reply components/nibegw/NibeGw.cpp:176]: Delay should not be any longer with the new code. Yhe old code would also call into the esphome logging in this codepath.

**Review by sins921 — APPROVED** [review]

**Pull request merged**